MigrationV3V4/Node/custom-recognizer-example/bot.js

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const { ActivityHandler } = require('botbuilder');
+const CustomRecognizer = require('./customRecognizer');
+
+class MyBot extends ActivityHandler {
+    constructor() {
+        super();
+        // See https://aka.ms/about-bot-activity-message to learn more about the message and other activity types.
+        this.onMessage(async (context, next) => {
+
+            // pass the response through custom recognizer to detect email addresses
+            const recognizer = new CustomRecognizer();
+            const recognizerResult = recognizer.recognize(context.activity.text);
+            const intent = recognizerResult.intents.length > 0 ? recognizer.getTopIntent(recognizerResult) : '';
+            
+            // respond according to the top intent provided by the recognizer
+            switch(intent) {
+              case 'SendEmail':
+                await context.sendActivity('Your response includes an email address. Would you like to contact us by email?');
+                break;
+              case 'SearchForEmailAddress':
+                await context.sendActivity('Your response includes an email address. Would you like to search for an email address?');
+                break;
+              case 'OpenEmailAccount':
+                await context.sendActivity('Your response includes an email address. Would you like to open an email account?');
+                break;
+              default:
+                await context.sendActivity('Your response includes an email address.'); 
```

Comment:
Since this is the default, won't it fire even if there is no email address in the response?

Reply:
Updated with a condition to only mention email if an intent was detected (ie the user input a valid email address).